docs/root/configuration/other_features/internal_listener.rst

```diff
@@ -0,0 +1,110 @@
+.. _config_internal_listener:
+
+Internal Listener
+=================
+
+How it works
+------------
+
+This extension contains 2 major components to add a listener with
+an :ref:`envoy internal address <envoy_v3_api_msg_config.core.v3.EnvoyInternalAddress>`
+and to create a client connection to that :ref:`listener <envoy_v3_api_msg_config.listener.v3.Listener>`
+
+envoy.bootstrap.internal_listener
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This bootstrap extension is required to support looking up the target listener via an
+:ref:`envoy internal address <envoy_v3_api_msg_config.core.v3.EnvoyInternalAddress>` on each worker threads.
+
+network.connection.client.envoy_internal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+It is a client connection factory. The factory is implicitly instantiated by the dispatcher to establish a client connection to an
+internal listener address.  This client connection factory is installed automatically when **envoy.bootstrap.internal_listerner** is specified.
+
+Example config
+--------------
+Below is a smallest static config that redirect tcp_proxy on port 19000 to the tcp_proxy binding to the internal address.
```

Comment:
not sure if `tcp_proxy` should be ` ``tcp_proxy`` ` or `TCP proxy`

Reply:
Seem all the rest in doc are "TCP proxy",  no inline literal. 
I update all 3 occurrences in this doc